cpp/ql/src/semmle/code/cpp/ir/IRSanity.ql

```diff
@@ -5,4 +5,4 @@
  * @id cpp/ir-sanity-check
  */
 
-import internal.IRSanityImpl
+import internal.aliased_ssa_ir.IRSanity
```

Comment:
Isn't this file and most other files in `ir` internal and should be moved to the `internal` dir? Some should probably be moved into the subdir of `internal` that's appropriate for the IR dialect they apply to. If users should only ever import `semmle.code.ir.IR`, then I'd say all other files are internal.

Reply:
There's no need for users to import e.g. `semmle.code.cpp.Function` rather than `cpp` either, is there?

I would say that `.internal.` should be for modules that are only privately imported, and that users should not import. i.e. we can freely change them without breaking user code.

Reply:
> I would say that `.internal.` should be for modules that are only privately imported, and that users should not import. i.e. we can freely change them without breaking user code.

I can see we've used "internal" with that meaning everywhere except in the pyrameterized modules of Java/C++ data flow and C++ IR. I agree it's a useful distinction and that we should probably reserve "internal" to mean what it's always meant.

I still see value in discouraging users from importing things like `semmle.code.cpp.ir.MemoryAccessKind` directly, though. I think of that file name as internal (subject to change between releases) even if the file contents is exposed somewhere in the namespace. Maybe a directory named "implementation" would be appropriate for such code?

Reply:
`semmle/code/cpp/ir` now contains only three modules: `IR.qll`, `PrintIR.ql`, and `IRSanity.ql`, all of which forward to the corresponding `implementation/aliased_ssa` versions. Everything else is now in either `implementation` or `internal`.